File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/contract_runme.m
contract

contract.test_preassert(1,2);
try
	contract.test_preassert(-1,0)
	error("Failed! Preassertions are broken")
catch
end_try_catch

contract.test_postassert(3);
try
	contract.test_postassert(-3);
	error("Failed! Postassertions are broken")
catch
end_try_catch

contract.test_prepost(2,3);
contract.test_prepost(5,-4);
try
	contract.test_prepost(-3,4);
	error("Failed! Preassertions are broken")
catch
end_try_catch

try
	contract.test_prepost(4,-10);
	error("Failed! Postassertions are broken")

catch
end_try_catch

f = contract.Foo();
f.test_preassert(4,5);
try
	f.test_preassert(-2,3);
	error("Failed! Method preassertion.")
catch
end_try_catch

f.test_postassert(4);
try
	f.test_postassert(-4);
	error("Failed! Method postassertion")
catch
end_try_catch

f.test_prepost(3,4);
f.test_prepost(4,-3);
try
	f.test_prepost(-4,2);
	error("Failed! Method preassertion.")
catch
end_try_catch

try
	f.test_prepost(4,-10);
	error("Failed! Method postassertion.")
catch
end_try_catch

contract.Foo_stest_prepost(4,0);
try
	contract.Foo_stest_prepost(-4,2);
	error("Failed! Static method preassertion")
catch
end_try_catch

try
	contract.Foo_stest_prepost(4,-10);
	error("Failed! Static method posteassertion")
catch
end_try_catch
	
b = contract.Bar();
try
	b.test_prepost(2,-4);
	error("Failed! Inherited preassertion.")
catch
end_try_catch


d = contract.D();
try
	d.foo(-1,1,1,1,1);
	error("Failed! Inherited preassertion (D).")
catch
end_try_catch
try
	d.foo(1,-1,1,1,1);
	error("Failed! Inherited preassertion (D).")
catch
end_try_catch
try
	d.foo(1,1,-1,1,1);
	error("Failed! Inherited preassertion (D).")
catch
end_try_catch
try
	d.foo(1,1,1,-1,1);
	error("Failed! Inherited preassertion (D).")
catch
end_try_catch
try
	d.foo(1,1,1,1,-1);
	error("Failed! Inherited preassertion (D).")
catch
end_try_catch


try
	d.bar(-1,1,1,1,1);
	error("Failed! Inherited preassertion (D).")
catch
end_try_catch
try
	d.bar(1,-1,1,1,1);
	error("Failed! Inherited preassertion (D).")
catch
end_try_catch
try
	d.bar(1,1,-1,1,1);
	error("Failed! Inherited preassertion (D).")
catch
end_try_catch
try
	d.bar(1,1,1,-1,1);
	error("Failed! Inherited preassertion (D).")
catch
end_try_catch
try
	d.bar(1,1,1,1,-1);
	error("Failed! Inherited preassertion (D).")
catch
end_try_catch

File: support_tools/swigwin-3.0.12/Examples/test-suite/perl5/contract_runme.pl
#!/usr/bin/perl
use strict;
use warnings;
use Test::More tests => 32;
BEGIN { use_ok('contract') }
require_ok('contract');

# adapted from ../python/contract_runme.py
{
	ok(contract::test_preassert(1,2), "good preassertion");
	eval { contract::test_preassert(-1) };
	like($@, qr/\bRuntimeError\b/, "bad preassertion");

	ok(contract::test_postassert(3), "good postassertion");
	eval { contract::test_postassert(-3) };
	like($@, qr/\bRuntimeError\b/, "bad postassertion");

	ok(contract::test_prepost(2,3), "good prepost");
	ok(contract::test_prepost(5,-4), "good prepost");
	eval { contract::test_prepost(-3,4); };
	like($@, qr/\bRuntimeError\b/, "bad preassertion");
	eval { contract::test_prepost(4,-10) };
	like($@, qr/\bRuntimeError\b/, "bad postassertion");
}
{
	my $f = contract::Foo->new();
	ok($f->test_preassert(4,5), "method pre");
	eval { $f->test_preassert(-2,3) };
	like($@, qr/\bRuntimeError\b/, "method pre bad");

	ok($f->test_postassert(4), "method post");
	eval { $f->test_postassert(-4) };
	like($@, qr/\bRuntimeError\b/, "method post bad");

	ok($f->test_prepost(3,4), "method prepost");
	ok($f->test_prepost(4,-3), "method prepost");
	eval { $f->test_prepost(-4,2) };
	like($@, qr/\bRuntimeError\b/, "method pre bad");
	eval { $f->test_prepost(4,-10) };
	like($@, qr/\bRuntimeError\b/, "method post bad");
}
{
	ok(contract::Foo::stest_prepost(4,0), "static method prepost");
	eval { contract::Foo::stest_prepost(-4,2) };
	like($@, qr/\bRuntimeError\b/, "static method pre bad");
	eval { contract::Foo::stest_prepost(4,-10) };
	like($@, qr/\bRuntimeError\b/, "static method post bad");
}
{
	my $b = contract::Bar->new();
	eval { $b->test_prepost(2,-4) };
	like($@, qr/\bRuntimeError\b/, "inherit pre bad");
}
{
	my $d = contract::D->new();
	eval { $d->foo(-1,1,1,1,1) };
	like($@, qr/\bRuntimeError\b/, "inherit pre D");
	eval { $d->foo(1,-1,1,1,1) };
	like($@, qr/\bRuntimeError\b/, "inherit pre D");
	eval { $d->foo(1,1,-1,1,1) };
	like($@, qr/\bRuntimeError\b/, "inherit pre D");
	eval { $d->foo(1,1,1,-1,1) };
	like($@, qr/\bRuntimeError\b/, "inherit pre D");
	eval { $d->foo(1,1,1,1,-1) };
	like($@, qr/\bRuntimeError\b/, "inherit pre D");

	eval { $d->bar(-1,1,1,1,1) };
	like($@, qr/\bRuntimeError\b/, "inherit pre D");
	eval { $d->bar(1,-1,1,1,1) };
	like($@, qr/\bRuntimeError\b/, "inherit pre D");
	eval { $d->bar(1,1,-1,1,1) };
	like($@, qr/\bRuntimeError\b/, "inherit pre D");
	eval { $d->bar(1,1,1,-1,1) };
	like($@, qr/\bRuntimeError\b/, "inherit pre D");
	eval { $d->bar(1,1,1,1,-1) };
	like($@, qr/\bRuntimeError\b/, "inherit pre D");
}

